src/components/dialog/dialog.tsx

```diff
@@ -0,0 +1,812 @@
+          capture: true
+        });
+      });
+    }
+  };
+
+  /**
+   * This observer watches the size of each border in the control to adjust the
+   * position of the invisible resize elements (edges and corners).
+   */
+  // eslint-disable-next-line @stencil-community/own-props-must-be-private
+  #setBorderSizeWatcher = () => {
+    if (!this.resizable || this.hidden) {
+      this.#removeBorderSizeWatcher();
+      return;
+    }
+
+    this.#borderSizeRAF ??= new SyncWithRAF();
+    this.#borderSizeObserver = new ResizeObserver(this.#updateBorderSizeRAF);
+
+    // Observe the size of the edges to know if the border
+    this.#borderSizeObserver.observe(this.el, { box: "border-box" });
+    this.#borderSizeObserver.observe(this.#resizeLayer);
+  };
+
+  #updateBorderSizeRAF = () => {
+    this.#borderSizeRAF.perform(this.#updateBorderSize);
+  };
+
+  #updateBorderSize = () => {
+    // - - - - - - - - - - - - - DOM read operations - - - - - - - - - - - - -
+    const computedStyle = getComputedStyle(this.el);
+
+    // - - - - - - - - - - - - - DOM write operations - - - - - - - - - - - - -
+    this.el.style.setProperty(
+      DIALOG_BORDER_BLOCK_START_SIZE,
+      computedStyle.borderBlockStartWidth
+    );
+
+    this.el.style.setProperty(
+      DIALOG_BORDER_BLOCK_END_SIZE,
+      computedStyle.borderBlockEndWidth
+    );
+
+    this.el.style.setProperty(
+      DIALOG_BORDER_INLINE_START_SIZE,
+      computedStyle.borderInlineStartWidth
+    );
+
+    this.el.style.setProperty(
+      DIALOG_BORDER_INLINE_END_SIZE,
+      computedStyle.borderInlineEndWidth
+    );
+  };
+
+  #removeBorderSizeWatcher = () => {
+    if (this.#borderSizeObserver) {
+      this.#borderSizeObserver.disconnect();
+      this.#borderSizeObserver = null; // Free the memory
+    }
+
+    this.#borderSizeRAF = null; // Free the memory
+  };
+
+  #fixDialogPosition = () => {
+    const dialogRect = this.#dialogRef.getBoundingClientRect();
+    const inlineStart = this.#isRTLDirection
+      ? document.documentElement.getBoundingClientRect().width -
+        (dialogRect.left + dialogRect.width)
+      : dialogRect.left;
+    const blockStart = dialogRect.top;
+
+    setProperty(
+      this.#dialogRef,
+      DIALOG_INLINE_START,
+      inlineStart - this.#draggedDistanceX
+    );
+    setProperty(
+      this.#dialogRef,
+      DIALOG_BLOCK_START,
+      blockStart - this.#draggedDistanceY
+    );
+  };
+
+  #unfixDialogPosition = () => {
+    this.#dialogRef.style.removeProperty(DIALOG_INLINE_START);
+    this.#dialogRef.style.removeProperty(DIALOG_BLOCK_START);
+  };
+
+  render() {
+    return (
+      <Host
+        class={{
+          "gx-dialog-header-drag": !this.hidden && this.allowDrag === "header",
+          [RESIZING_CLASS]: this.resizing
+        }}
+      >
+        <dialog
+          class={this.showHeader ? "dialog--header" : null}
+          aria-labelledby={this.caption ? "heading" : null}
+          part="dialog"
```

Comment:
```suggestion
          aria-labelledby={this.caption ? "heading" : null}
          class={this.showHeader ? "dialog--header" : null}
          part="dialog"
```

Reply:
should aria-* be allways the firstaattribute, or is it because alphabetical order?